pkg/granted/credentials.go

```diff
@@ -0,0 +1,441 @@
+package granted
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/bigkevmcd/go-configparser"
+
+	"github.com/common-fate/granted/internal/build"
+	"github.com/common-fate/granted/pkg/cfaws"
+	"github.com/common-fate/granted/pkg/securestorage"
+	"github.com/common-fate/granted/pkg/testable"
+	"github.com/urfave/cli/v2"
+)
+
+var CredentialsCommand = cli.Command{
+	Name:        "credentials",
+	Usage:       "Manage secure IAM credentials",
+	Subcommands: []*cli.Command{&AddCredentialsCommand, &ImportCredentialsCommand, &UpdateCredentialsCommand, &ListCredentialsCommand, &ClearCredentialsCommand, &ExportCredentialsCommand},
+}
+
+// promptProfileName checks the first argument for a profile name else prompts for a name
+func promptProfileName(c *cli.Context) (string, error) {
+	profileName := c.Args().First()
+	if profileName == "" {
+		in := survey.Input{Message: "Profile name: "}
+		err := testable.AskOne(&in, &profileName)
+		if err != nil {
+			return "", err
+		}
+	}
+	return profileName, nil
+}
+
+var AddCredentialsCommand = cli.Command{
+	Name:  "add",
+	Usage: "Add IAM credentials to secure storage",
+	Action: func(c *cli.Context) error {
+		profileName, err := promptProfileName(c)
+		if err != nil {
+			return err
+		}
+
+		// validate the the profile does not already exist
+		profiles, err := cfaws.LoadProfiles()
+		if err != nil {
+			return err
+		}
+		if profiles.HasProfile(profileName) {
+			return fmt.Errorf("a profile with name %s already exists", profileName)
+		}
+
+		credentials, err := promptCredentials()
+		if err != nil {
+			return err
+		}
+
+		// store the credentials in secure storage
+		secureIAMCredentialStorage := securestorage.NewSecureIAMCredentialStorage()
+		err = secureIAMCredentialStorage.StoreCredentials(profileName, credentials)
+		if err != nil {
+			return err
+		}
+		err = updateOrCreateProfileWithCredentialProcess(profileName)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Saved %s to secure storage", profileName)
```

Comment:
A nitpick on the Printf statements: all of them in this file need a `\n` at the end so that they print a newline properly